docs/DATA_MODEL.md

```diff
@@ -0,0 +1,299 @@
+* `description:lang=de:format=date`
+* `description:lang=en`
+* `description:lang=en:format=date`
+
+For the query `"description" contains "my search text"` Search Service will first select all PropertyVariants which
+match the given KeyFilter `description`, which are all of the above. Then it will discard all with an invalid format
+Variant, which leaves us with the PropertyVariants `description`, `description:lang=de` and `description:lang=en`. It
+will now evaluate the `contains` query against the value of those three PropertyVariants, and if any match the query
+will evaluate to true.
+
+## Variants and the HTML-Publisher
+
+For displaying properties with different Formats the HTML-Publisher follows
+the [Semantic Conventions](SEMANTIC_CONVENTIONS.md) and tries to find the Variant with the highest priority.
+
+If the user has selected a language (feature pending :P ) the Publisher will preferably select a PropertyVariant with
+the selected language.
+If none are available (or the user has no language selected) it will fall back to the default language (aka
+PropertyVariants without any language).
+If none are available it will select any PropertyVariant with a valid Format.
+
+For example, given following PropertyVariants:
+
+* `description`
+* `description:lang=de:format=date`
+* `description:lang=en`
+* `description:lang=en:format=number`
+
+If the Publisher wants to show some property with the format `text` (aka no format Variant) and the user has selected
+the language `de`, the Publisher
+would first look at all PropertyVariants with `lang=de`. There is only one, `description:lang=de:format=date`, but it
+has the wrong format so the Publisher will try again with the default language, aka `lang=`.
+The only PropertyVariant without language is `description`, which has the correct format and is selected.
+
+For a second example, lets say the user still has selected the language `de` but the Publisher needs the format `number`
+to be able to show the property.
+In this case the first step is still the same, but the second step where we look at the PropertyVariant `description`
+now also fails because it has the wrong format.
+Now in the third step the Publisher will ignore the language format and will look at all PropertyVariants to find a
+valid one. In this
+example there is only one so it will select `description:lang=en:format=number`. If there would be more than one valid
+Variant with the same format priority, it will sort them alphabetically and select the first one. (Like
+the [KeyFilters](#keyfilters) specify.)
+
+## Variants and the iCal-Publisher
+
+Similar to the HTML-Publisher, only that it cannot render Markdown and will ignore this format.
+
+# Known Variants
+
+## Language Variant
+
+Has the VariantName "lang" and the value is a two-letter language abbreviation, like "en", "de", "fr", ...
+Denotes the language the value is in. Without a language variant it is assumed that this is the "default" language and
```

Comment:
what is the default lang as fallback? can I request multiple lang variantNames chained via ","?

Reply:
"what is the default lang as fallback?" it is not a language, it is meant as a default text compared to variants who explicitly set a language.